lib/smack/SmackRep.cpp

```diff
@@ -239,16 +340,24 @@ const Expr* SmackRep::lit(const llvm::Value* v) {
       Expr::lit(exponentPart));
 
   } else if (llvm::isa<llvm::ConstantPointerNull>(v))
-    return Expr::lit(0, width);
+    return Expr::lit(0, MAXSIZE);
```

Comment:
How about defining a Boogie constant `const null: ref` which is axiomatized to be `null == 0` in the non-bitvector case, and `null == 0bvXX` in the bitvector case, depending on actual pointer size.